### ISSUE TITLE
BowInfinityFix-1.8-rv1 Server Hangs

### ISSUE THREAD

**Issue by cauliflower69** [SpongePowered/SpongeForge#496]
No crash report but here is the server log

http://pastebin.com/mXv3Xyf9

Works fine with just Forge

**Comment by gabizou** [comment]
This is an issue for @zml2008. From a cursory glance, one of the mods is implementing their own `ModContainer` and thus, we can't call `getId()`. @cauliflower69 can you add one mod at a time (starting with SpongeForge) and see which mod causes the server to crash?

For reference, the important bit of stack trace:
```
[13:25:38] [main/ERROR] [LaunchWrapper/]: Unable to launch
java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_66]
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_66]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_66]
    at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_66]
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_66]
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_66]
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_66]
    at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_66]
    at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.run(ServerLaunchWrapper.java:43) [ForgeMod.jar:?]
    at net.minecraftforge.fml.relauncher.ServerLaunchWrapper.main(ServerLaunchWrapper.java:12) [ForgeMod.jar:?]
Caused by: java.lang.AbstractMethodError: Method net/minecraftforge/fml/common/MinecraftDummyContainer.getId()Ljava/lang/String; is abstract
    at net.minecraftforge.fml.common.MinecraftDummyContainer.getId(MinecraftDummyContainer.java) ~[MinecraftDummyContainer.class:?]
    at net.minecraft.command.ServerCommandManager.func_71560_a(SourceFile:103) ~[bd.class:?]
    at net.minecraft.command.ServerCommandManager.<init>(SourceFile:15) ~[bd.class:?]
    at net.minecraft.server.MinecraftServer.func_175582_h(MinecraftServer.java:175) ~[MinecraftServer.class:?]
    at net.minecraft.server.MinecraftServer.<init>(MinecraftServer.java:166) ~[MinecraftServer.class:?]
    at net.minecraft.server.dedicated.DedicatedServer.<init>(DedicatedServer.java:56) ~[ko.class:?]
    at net.minecraft.server.MinecraftServer.main(MinecraftServer.java:1475) ~[MinecraftServer.class:?]
    ... 12 more
```
Which
```
    at net.minecraft.command.ServerCommandManager.func_71560_a(SourceFile:103) ~[bd.class:?]
```
points to: [MixinServerCommandManager](https://github.com/SpongePowered/SpongeCommon/blob/master/src/main/java/org/spongepowered/common/mixin/core/server/MixinServerCommandManager.java#L103).

**Comment by cauliflower69** [comment]
I was adding each mod individually. This is a WIP pack on Technic called Requiem 189. The server worked and booted fine with Forge. 

I tried to use sponge and had many crashes so I added each mod at a time and posted the crashes here as they came up. 

Out of 90 mods less than five had issues.

**Comment by gabizou** [comment]
> I tried to use sponge and had many crashes so I added each mod at a time and posted the crashes here as they came up.

Could you provide these logs and specify which mod being added had which crash log?

**Comment by cauliflower69** [comment]
I have done that already with each submission I made today. This mod did not create a crash report that's why I gave you the server log

**Comment by Zidane** [comment]
@cauliflower69 

We understand but we cannot know which mod is the culprit from this log. What was the LAST mod you added to your pack (with SpongeForge installed) that caused you to get this stacktrace on server start?

**Comment by ryantheleach** [comment]
I'm guessing it's BowInfinityFix-1.8-rv1 from the title.

**Comment by cauliflower69** [comment]
@ryantheleach Thanks

**Comment by simon816** [comment]
@gabizou This is not a problem with the mod implementing their own `ModContainer`
The crash is on `MinecraftDummyContainer` 
> Caused by: java.lang.AbstractMethodError: Method net/minecraftforge/fml/common/MinecraftDummyContainer.getId()Ljava/lang/String; is abstract

We [mix-in to DummyModContainer](https://github.com/SpongePowered/SpongeForge/blob/master/src/main/java/org/spongepowered/mod/mixin/core/fml/common/MixinModContainer.java) which is the super class of `MinecraftDummyContainer`


The _actual_ problem is that BoxInfinityFix [force-loads `DummyModContainer`](https://github.com/Parker8283/BowInfinityFix/blob/master/src/main/java/com/github/parker8283/bif/BowInfinityFix.java#L18) before our mixin system has initialized.
This is the exact same issue that we had with OpenComputers (https://github.com/SpongePowered/SpongeForge/issues/362)
The only thing we can do is get the mod to fix it their end (like with OC https://github.com/MightyPirates/OpenComputers/issues/1485)

**Comment by simon816** [comment]
Issue reported, https://github.com/Parker8283/BowInfinityFix/issues/1, check there for progress.
This spongeforge issue will remain open until a fix has been confirmed.

**Comment by Parker8283** [comment]
Give [rv2](http://minecraft.curseforge.com/projects/bow-infinity-fix/files/2276241) a shot. Should fix the problem (and now you can run it on 1.8.9...which means if I was holding you back on 1.8, I'm sorry :P)

**Comment by simon816** [comment]
I can confirm this is fixed in rv2
